### PR TITLE
Added provider parameter to find_resource 

### PR DESCRIPTION
**Pull request by m4-miranda**
As discussed in comments of #997,
I've added a provider paramter to find_resource method. I've updated the calls to find_resource method in passwords controller:
```golang 
@resource = find_resource(:uid, @email, provider)
```
sessions controller:
```golang 
@resource = find_resource(field, q_value, provider)
 ```
Unlocks controller, I assume, would require an 'email' provider hence I've hardcoded the paramter there:
```golang
@resource = find_resource(:email, @email, 'email')
```
P.S I'm fairly new to devise_token_auth and devise in general so not sure about the best practices etc.
@MaicolBen @zachfeldman

### CODE REVIEW

File: app/controllers/devise_token_auth/concerns/resource_finder.rb [hunk 1]
```diff
@@ -12,10 +12,9 @@ def get_case_insensitive_field_from_resource_params(field)
     q_value
   end
 
-  def find_resource(field, value)
+  def find_resource(field, value, provider)
```
Comment (MaicolBen) [inline app/controllers/devise_token_auth/concerns/resource_finder.rb]: I don't understand why the parameter, this won't call the method

Comment (m4-miranda) [reply app/controllers/devise_token_auth/concerns/resource_finder.rb]: Hi, could you help me out with how to go about overriding the concern in my project that utilises this repo? I would write a concern that extends `DeviseTokenAuth::Concerns::ResourceFinder` with a ```def provider``` and override the controllers to `include customResourceFinder`. Is this right?

Comment (MaicolBen) [reply app/controllers/devise_token_auth/concerns/resource_finder.rb]: I think that won't work, you need to create a concern with `def provider` in there and add that concern in each override controller. But if you need to add it to the registrations controller, you add it there. But you need to add PR here with only this:
screen shot 2017-10-27 at 12 56 28 pm

File: app/controllers/devise_token_auth/passwords_controller.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def create
       end
 
       @email = get_case_insensitive_field_from_resource_params(:email)
-      @resource = find_resource(:uid, @email)
+      @resource = find_resource(:uid, @email, provider)
```
Comment (MaicolBen) [inline app/controllers/devise_token_auth/passwords_controller.rb]: You forgot to remove from here

**Pull request merged**